src/org/loklak/api/search/WordpressCrawlerService.java

```diff
@@ -57,87 +72,90 @@ public JSONObject getDefaultPermissions(BaseUserRole baseUserRole) {
-
-		String[][] blogPosts = new String[100][4];
-
-		// blogPosts[][0] = Blog Title
-		// blogPosts[][1] = Posted On
-		// blogPosts[][2] = Author
-		// blogPosts[][3] = Blog Content
-
-		Integer numberOfBlogs = 0;
-		Integer iterator = 0;
-
-		try {
-			blogHTML = Jsoup.connect(blogURL).userAgent("Mozilla").get();
-		} catch (IOException e) {
-			e.printStackTrace();
-		}
-
-		articles = blogHTML.getElementsByTag("article");
-
-		iterator = 0;
-		for (Element article : articles) {
-
-			articleList_title = article.getElementsByClass("entry-title");
-			for (Element blogs : articleList_title) {
-				blogPosts[iterator][0] = blogs.text().toString();
-			}
-
-			articleList_dateTime = article.getElementsByClass("posted-on");
-			for (Element blogs : articleList_dateTime) {
-				blogPosts[iterator][1] = blogs.text().toString();
-			}
-
-			articleList_author = article.getElementsByClass("byline");
-			for (Element blogs : articleList_author) {
-				String author = blogs.text().toString();
-				author = author.substring(author.indexOf(' ') + 1);
-				blogPosts[iterator][2] = author;
-			}
-
-			articleList_content = article.getElementsByClass("entry-content");
-			for (Element blogs : articleList_content) {
-				blogPosts[iterator][3] = blogs.text().toString();
-			}
-
-			iterator++;
-
-		}
-
-		numberOfBlogs = iterator;
-
-		JSONArray blog = new JSONArray();
-
-		for (int k = 0; k < numberOfBlogs; k++) {
-			JSONObject blogpost = new JSONObject();
-			blogpost.put("blog_url", blogURL);
-			blogpost.put("title", blogPosts[k][0]);
-			blogpost.put("posted_on", blogPosts[k][1]);
-			blogpost.put("author", blogPosts[k][2]);
-			blogpost.put("content", blogPosts[k][3]);
-			blog.put(blogpost);
-		}
-		
-		SusiThought json = new SusiThought();
-		json.setData(blog);
-		return json;
-	
-	}
+    protected String prepareSearchUrl(String type) {
+        return this.query;
+    }
+
+    protected void setParam() {
+        if ("".equals(this.getExtraValue("url"))) {
+            this.setExtraValue("url", this.query);
+        } else {
+            this.query  = this.getExtraValue("url");
+        }
+    }
+
+   protected Post scrape(BufferedReader br, String type, String url) {
+        Post typeArray = new Post(true);
+        typeArray.put(this.scraperName, this.crawlWordpress(this.query, br).toArray());
+        return typeArray;
+    }
+
+
+    public Timeline2 crawlWordpress(String blogURL, BufferedReader br) {
+
+        Document blogHTML = null;
+        Post blogPost = null;
+        Elements articles = null;
+        Elements articleList_title = null;
+        Elements articleList_content = null;
+        Elements articleList_dateTime = null;
+        Elements articleList_author = null;
+
+        Timeline2 blogList = new Timeline2(this.order);
+
+        try {
+	        blogHTML = Jsoup.parse(bufferedReaderToString(br));
+        } catch (IOException e) {
+	        DAO.trace(e);
+        }
+
+        String[][] blogPostList = new String[100][4];
```

Comment:
I just configured BaseScraper and Post. this isn't related to PR. I will fix it.